lektor/imagetools.py

```diff
@@ -22,30 +22,32 @@
 datetime.strptime("", "")
 
 
-class ThumbnailMode(IntEnum):
-    FIT, CROP, STRETCH = range(1, 4)
+class ThumbnailMode(Enum):
+    FIT = "fit"
+    CROP = "crop"
+    STRETCH = "stretch"
+
+    DEFAULT = "fit"
 
     @property
     def label(self):
         """The mode's label as used in templates."""
-        # our constants are uppercase with underscores,
-        # while template uses lowercase and dashes.
-        return self.name.lower().replace("_", "-")  # pylint: disable=no-member
+        warnings.warn(
+            "ThumbnailMode.label is deprecated. (Use ThumbnailMode.value instead.)",
+            DeprecationWarning,
+        )
+        return self.value
```

Comment:
Well, it still works.  It just issues a DeprecationWarning.  (And python's default warnings filter is set to ignore DeprecationWarning, except from `__main__`, so your typical user won't see the warning anyway, I think.)
It could be made a PendingDeprecationWarning,  but I'm not sure that's necessary.

Reply:
Fair enough, I guess they could catch it using the warnings module.

Reply:
Okay, I'm leaving this.

To be clear, old code (which relies on `ThumbnailMode.label` and `.from_label`) will still work fine.  (At least, that is the intention.)  It will just issue a `DeprecationWarning`.  The [default setting of Python’s warningsfilter](https://docs.python.org/3/library/warnings.html#default-warning-filter) is to ignore `DeprecationWarning`s, so really, the normal user will not notice any change — no messages on the console, nothing.  Nobody should have to explicitly catch anything for old code to continue working.

The purpose of the warning is to notify authors of third-party software (plugins) that planned future changes in the Lektor code will break their software.  To that end, pytest does (by default) enable and report about `DeprecationWarning`s, but these still will not cause test failure — they'll just do their job to warn the developer that at some point the functionality will go away.

The other option, `PendingDeprecationWarning`, as I understand it, is an even gentler hint.  `PendingDeprecationWarning` signals that at some future time (e.g. the release of Lektor 4.0), the warning will be converted to a `DeprecationWarning` (after which, after even more time — e.g., Lektor 5.0 — the feature will be removed.)   However, `PendingDeprecationWarning` and `DeprecationWarning` are handled almost identically, both by the default settings of the warningsfilter (which hides both), and by pytest (which reports both) — so, other than semantics, there's not much practical difference.

(There's even more discussion in [PEP565](https://www.python.org/dev/peps/pep-0565/#additional-use-case-for-futurewarning).)